app/controllers/application_controller.rb

```diff
@@ -16,6 +16,19 @@ class ApplicationController < ActionController::Base
 
   protected
 
+  def current_admin_user
+    return current_user if current_user.admin?
+    return nil
```

Comment:
there is no need for a return nil here as it will return nil if the condition is not met.